src/v/v8_engine/api.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "v8_engine/api.h"
+
+#include "config/configuration.h"
+#include "syschecks/syschecks.h"
+
+namespace v8_engine {
+
+api::api(ss::alien::instance& instance)
+  : _executor(instance, config::shard_local_cfg().executor_queue_size()) {}
+
+ss::future<> api::start() {
+    if (is_enabled()) {
+        co_await syschecks::systemd_message("Creating v8_engine::api");
+        co_await _executor.start();
+        co_await _script_dispatcher.start(std::ref(_executor));
+    }
+}
+
+ss::future<> api::stop() {
+    if (is_enabled()) {
+        co_await _script_dispatcher.stop();
+        co_await _executor.stop();
+    }
+}
+
+ss::future<> api::insert_code(coproc::script_id id, iobuf code) {
+    if (!is_enabled()) {
```

Comment:
I think you can omit these checks outside of start() & stop(), if the code reaches here and it shouldn't have, it will crash because it is accessing uninitialized services, which IMO is more desirable. Otherwise it seems like the code path could come here but in these cases, nothing is done anyway